TestResultSummaryService/plugins/DataManagerPerf.js

```diff
@@ -0,0 +1,71 @@
+    if ( task.hasChildren && !task.aggregateInfo) {
+        const parentId = task._id;
+        const childBuildList = await testResultsDB.getData({parentId}).toArray();
+        const parentAggregationStructure = [];
+        let bName, bVariant, bProduct;
+        const childrenAggData = [];
+        const metricsCollection = {};
+        //loop into the child build list
+        for ( let childBuild of childBuildList ) {
+            //loop into the each child's tests info.
+            if (Array.isArray(childBuild.tests)) {
+                for ( let {benchmarkName, benchmarkVariant, benchmarkProduct, testData} of childBuild.tests){
+                    bName = benchmarkName;
+                    bVariant = benchmarkVariant;
+                    bProduct = benchmarkProduct;
+                    if ( Array.isArray(testData.metrics) ) {
+                        for ( let {name, value} of testData.metrics ){
+                            metricsCollection[name] = metricsCollection[name] || [];
+                            if ( Array.isArray(value) ) {
+                                metricsCollection[name] = metricsCollection[name].concat(value);
+                            }
+                            //remove the bad data, ie: null.
+                            metricsCollection[name] = metricsCollection[name].filter (n => n);
+                        }
+                    }
+                }
+            }
+        }
+        // calculate the aggregate data
+        Object.keys( metricsCollection ).forEach( function(key) {
+            const meanMetric = math.mean(metricsCollection[key]);
+            const maxMetric = math.max(metricsCollection[key]);
+            const minMetric = math.min(metricsCollection[key]);
+            const medianMetric = math.median(metricsCollection[key]);
+            const stddevMetric = math.std(metricsCollection[key]);
+            const ciMetric = BenchmarkMath.confidence_interval(metricsCollection[key]);
+            childrenAggData.push({
+                name: key, 
+                value: { 
+                    mean: meanMetric,
+                    max: maxMetric,
+                    min: minMetric,
+                    median: medianMetric,
+                    stddev: stddevMetric,
+                    CI: ciMetric,
+                    iteration: childBuildList.length
+                }
+            });
+        })
+        //add aggregate info for master node and update it in database
+        parentAggregationStructure.push({
+            benchmarkName: bName,
+            benchmarkVariant: bVariant,
+            benchmarkProduct: bProduct,
+            metrics: childrenAggData
+        });
+        const { _id, ...value } = task;
+        const criteria = { _id: new ObjectID( _id ) };
+        const update = {
+                ...value
+        };
+        update.aggregateInfo = parentAggregationStructure;
+        const result = await testResultsDB.update( criteria, { $set: update } );
```

Comment:
Please remove line 64 - 68. Rename `parentAggregationStructure ` to `aggregateInfo `
We just need one line:
`const result = await testResultsDB.update( criteria, { $set: {aggregateInfo} });`